flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -120,6 +132,34 @@ public void closeOpenChannelConnections(ConnectionID connectionId) {
 		}
 	}
 
+	void connectChannelWithRetry(ConnectingChannel connectingChannel, ConnectionID connectionId)
+		throws IOException, InterruptedException {
+		int count = 0;
+		Exception exception = null;
+		do {
+			try {
+				connectingChannel = new ConnectingChannel(connectionId, this);
```

Comment:
this can be done outside of loop, to not create it every retry.

Reply:
It is intentionally to create new instance each so that the old status such exception is cleaned.

Reply:
I see it is already out of loop in latest changes.